### PR TITLE
A swifty PR suggested for issue #1331

### PR DESCRIPTION
**Pull request by dafnapension**
closes: https://github.com/IBM/unitxt/issues/1331

Added a test to identify cases of more than one prepare file generating the same catalog file.
Consistency test works in granularity of prepare-files (can not identify two or more generations of same catalog file, within same prepare file; only across different prepare-files). But for now, and because typically, not too many `add_to_catalog` within same prepare_file, and hence can be manually tested for uniqueness, this PR may suffice.

### CODE REVIEW

File: prepare/processors/processors.py [hunk 1]
```diff
@@ -171,31 +171,6 @@
     overwrite=True,
 )
 
-add_to_catalog(
```
Comment (elronbandel) [inline prepare/processors/processors.py:174]: Why did you delete these?

**Review by elronbandel — CHANGES_REQUESTED** [review]

File: prepare/processors/processors.py (hunk 1 above)
Comment (elronbandel) [reply prepare/processors/processors.py:174]: I mean where are they added somewhere else? in the same file?

**Comment by dafnapension** [comment]
Here is an excerpt from `processors.py` of current `main` branch, please notice the line numbers and contents:

image

Now please notice the excerpt from same file of same branch, different lines, immediately following the previous ones, but same contents:

image

Someone pasted twice by mistake, and I removed once.

**Comment by dafnapension** [comment]
On this occasion, 'while the hood is open', I added a test to catelog-consistency, to verify that not two prepare-files generate the same catalog entry (overwriting each other).  There is still no test to catch errors like the one triggered this PR: overwriting in within same prepare-file, because the automatic test works in granularity of whole prepare files.
I commented that errors like this one (that triggered this PR) can be caught (evidently, not too obviously..) by the eye..

**Review by elronbandel — APPROVED** [review]

**Pull request merged**